packages/@o3r/extractors/src/core/builder.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  BuilderContext,
+  BuilderOutput,
+  createBuilder
+} from '@angular-devkit/architect';
+import { getEnvironmentInfo } from '@o3r/schematics';
+import { performance } from 'node:perf_hooks';
+
+/**
+ * Wrapper method of `createBuilder` from `@angular-devkit-architect`
+ * to handle metrics around the builder run
+ * @param builderFn
+ */
+export const createBuilderWithMetrics = <S, O extends BuilderOutput = BuilderOutput>(
+  builderFn: (opts: S, ctx: BuilderContext) => O | Promise<O>
+) => createBuilder<S, O>(
```

Comment:
To be future proof it wouls be interesting to be able to chain the `createBuilder`s as you have done for the schematics.
Today you Builder (or schematics) builder wrap the builder (or schematic) creation to add metric but it may not be the only wrapper to apply to builders/schematics.
The current implementation is working well for schematics but not builders